File: src/leetcode/2025/12/Solution342.cpp
#include <iostream>

class Solution342
{
public:
    bool isPowerOfFour(int n)
    {
        if (n < 1)
        {
            return false;
        }
        while (n % 4 == 0)
        {
            n /= 4;
        }
        return n == 1;
    }
};

int main()
{
    Solution342 solution;

    std::cout << solution.isPowerOfFour(16) << std::endl; // true
    std::cout << solution.isPowerOfFour(5) << std::endl;  // false
    std::cout << solution.isPowerOfFour(1) << std::endl;  // true
    std::cout << solution.isPowerOfFour(64) << std::endl; // true
    return 0;
}

File: src/leetcode/2025/12/Solution342.java
public class Solution342 {
    public boolean isPowerOfFour(int n) {
        if (n < 1) {
            return false;
        }
        while (n % 4 == 0) {
            n /= 4;
        }
        return n == 1;
    }

    public static void main(String[] args) {
        Solution342 solution = new Solution342();
        System.out.println(solution.isPowerOfFour(16)); // true
        System.out.println(solution.isPowerOfFour(5)); // false
        System.out.println(solution.isPowerOfFour(1)); // true
        System.out.println(solution.isPowerOfFour(-4)); // false
        System.out.println(solution.isPowerOfFour(64)); // true
        System.out.println(solution.isPowerOfFour(20)); // false
    }
}
